### PR TITLE
fix: ASC-2826 - allow user to update displayName & discription

### PR DESCRIPTION
**Pull request by Cijin**
### Checklist

- [x] There is an associated JIRA issue (**required**)
- [x] The title of your PR is formatted properly (**required**)
- [x] Code is up-to-date with the `develop` branch (**required**)

- [x] You've successfully run your unit tests locally
- [ ] There are new or updated unit tests validating the change

### Short description
* fix:change default avatarFileId from empty to null
* Reason: the sdk validation will fail if avatarFileId is empty
* Which leads to an undesirable result where user cannot update
  description or displayName if there is no avatarFileId set
* feat:add util to filter dirty values from form
* feat:filter out dirtyValues from UserProfileForm
* Update src/helpers/utils.js
* chore: asc-2826 remove dirtyValues helper

Co-authored-by: Pawarit (Nhui) Chongkrichug <nightwsh58@gmail.com>

### CODE REVIEW

File: src/social/components/UserProfileForm/index.js [hunk 1]
```diff
@@ -40,18 +40,14 @@ const UserProfileForm = ({ user, onSubmit, className }) => {
...
-      description: '',
+      avatarFileId: user.avatarFileId ?? null,
     },
   });
 
   const description = watch('description');
   const displayName = watch('displayName');
 
-  return (
+  return -(
```
Comment (serhii-ivanenko-amity) [inline src/social/components/UserProfileForm/index.js]: not sure if this char should be here

**Review by serhii-ivanenko-amity — CHANGES_REQUESTED** [review]

**Review by serhii-ivanenko-amity — DISMISSED** [review]

File: .github/workflows/dev.yaml [hunk 2]
```diff
@@ -12,7 +12,8 @@ jobs:
       url: https://ui-kit-open-source.dev.amity.co/${{ github.event.pull_request.number }}
 
     env:
-      STORYBOOK_API_ENDPOINT: ${{ secrets.STORYBOOK_API_ENDPOINT }}
+      NPM_FONT_AWESOME_TOKEN: ${{ secrets.NPM_FONT_AWESOME_TOKEN }}
```
Comment (serhii-ivanenko-amity) [inline .github/workflows/dev.yaml]: we do not need this one here. It's open source ui-kit, it not use font awesome

File: .github/workflows/dev.yaml [hunk 3]
```diff
@@ -12,7 +12,8 @@ jobs:
       url: https://ui-kit-open-source.dev.amity.co/${{ github.event.pull_request.number }}
 
     env:
-      STORYBOOK_API_ENDPOINT: ${{ secrets.STORYBOOK_API_ENDPOINT }}
```
Comment (serhii-ivanenko-amity) [inline .github/workflows/dev.yaml]: why this removed ?

File: .github/workflows/staging.yaml [hunk 4]
```diff
@@ -14,7 +14,8 @@ jobs:
       url: https://ui-kit-open-source.staging.amity.co/
 
     env:
-      STORYBOOK_API_ENDPOINT: ${{ secrets.STORYBOOK_API_ENDPOINT }}
+      NPM_FONT_AWESOME_TOKEN: ${{ secrets.NPM_FONT_AWESOME_TOKEN }}
```
Comment (serhii-ivanenko-amity) [inline .github/workflows/staging.yaml]: same

File: .github/workflows/staging.yaml [hunk 5]
```diff
@@ -14,7 +14,8 @@ jobs:
       url: https://ui-kit-open-source.staging.amity.co/
 
     env:
-      STORYBOOK_API_ENDPOINT: ${{ secrets.STORYBOOK_API_ENDPOINT }}
```
Comment (serhii-ivanenko-amity) [inline .github/workflows/staging.yaml]: same

File: src/core/providers/UiKitProvider/index.js [hunk 6]
```diff
@@ -128,7 +128,7 @@ const UiKitProvider = forwardRef(
 
 UiKitProvider.propTypes = {
   apiKey: PropTypes.string.isRequired,
-  apiEndpoint: PropTypes.string.isRequired,
```
Comment (serhii-ivanenko-amity) [inline src/core/providers/UiKitProvider/index.js]: why this not required ? is there any default value ?

**Review by serhii-ivanenko-amity — CHANGES_REQUESTED** [review]

**Comment by Cijin** [comment]
Is this the reason for failing action?
https://github.com/actions/runner/discussions/1380

File: .github/workflows/dev.yaml (hunk 3 above)
Comment (Va1tra) [reply .github/workflows/dev.yaml]: The recommended way to configure the endpoint is to set the region.

**Review by serhii-ivanenko-amity — DISMISSED** [review]

**Pull request closed without merging**